hpx/lcos/local/receive_buffer.hpp

```diff
@@ -147,6 +147,13 @@ namespace hpx { namespace lcos { namespace local
             entry->set_value(std::move(val));
         }
 
+        // explicitly clear all sored data
+        void clear_all()
+        {
+            std::lock_guard<mutex_type> l(mtx_);
+            buffer_map_.clear();
```

Comment:
The receive buffer itself doesn't know about future<id_type>. I think it should be a error if the receive buffer leaves non collected data behind. In the case of the stencil example, this should be stuff sent on the last time step but never received, iiuc.